tasks/DroneFlightControlTask.cpp

```diff
@@ -604,11 +608,21 @@ bool DroneFlightControlTask::teardownSubscription(const int pkgIndex)
 bool DroneFlightControlTask::canTakeControl(Telemetry::SDKInfo const& control_device)
 {
     /** This check whether the SDK is controlling the drone or if the control device
-     * changes to the Remote Controller.
+     * changes to the Remote Controller, and outputs the state.
      *
      * 0 - RC
      * 1 - Mobile app
      * 2 - Serial
      */
+    auto dji_flight_status =
+        mVehicle->subscribe->getValue<Telemetry::TOPIC_STATUS_FLIGHT>();
+    mStatus.control_device = static_cast<ControlDevice>(control_device.deviceStatus);
+    mStatus.device_flight_status =
+        static_cast<DeviceFlightStatus>(control_device.flightStatus);
+    _status.write(mStatus);
+    drone_control::FlightStatus flight_status =
+        static_cast<drone_control::FlightStatus>(dji_flight_status);
+    _flight_status.write(flight_status);
+
```

Comment:
Turn this into another function and call it both here and at the `updateHook`. The way it is rn, we would only output telemetry data when we were checking if the sdk could take control.

Reply:
definitely. Thanks